backend/src/Edna/Analysis/Infra.hs

```diff
@@ -50,20 +44,18 @@ instance Exception PythonError where
 callPythonAnalysis ::
   (ToJSON inp, FromJSON out) => FilePath -> inp -> Edna out
 callPythonAnalysis pyPath request = do
-  let requestString = decodeUtf8 $ encode request
-  logDebug $ toText $ "Python data request: " <> requestString
+  let requestJSON = encode request
+  logDebug $ "Python data request: " <> decodeUtf8 requestJSON
   analysisDir <- liftIO $
     fromMaybe (".." </> "analysis") <$> lookupEnv "EDNA_ANALYSIS_DIR"
-  (exitCode, out, err) <- liftIO $ readCreateProcessWithExitCode
-    (proc "python3" [pyPath, requestString])
-    { cwd = Just analysisDir, std_out = CreatePipe, std_err = CreatePipe}
-    ""
-  let isSuccess = exitCode == ExitSuccess
-  -- Logging with debug severity in case of success.
-  -- Not logging in case of failure because the whole stderr will be provided
-  -- in the thrown exception.
-  when (not (null err) && isSuccess) $
-    logDebug $ "Unexpected python stderr: " <> toText err
-  unless isSuccess $ throwM $ PyExitFailure exitCode err
-  logDebug $ toText $ "Python data response: " <> out
-  either (throwM . PyInvalidFormat . toText) pure $ eitherDecode (encodeUtf8 out)
+  let processConfig =
+        setStdin (byteStringInput requestJSON) $
+        setWorkingDir analysisDir $
+        proc "python3" [pyPath]
+  -- @readProcess_@ automatically captures stdout and stderr.
+  -- It also checks the exit code and throws an exception if it's not 0.
```

Comment:
We were throwing an exception in case of non-zero exit code before this PR as well, the only difference is that we were throwing `PyExitFailure` which doesn't exist anymore. After recent changes in Python code we handle bad cases there explicitly and report them as `"status": "ERROR blabla"`, so `ic50_analysis.py` should exit with 0 for any input data (as long as it can be parsed). If it returns non-zero, it's a bug and I think throwing an exception is the best option in this case (it will be reported as internal server error and it's indeed an internal error, it's a bug in code).